noxfile.py

```diff
@@ -203,6 +203,10 @@ def build_release(session):
     session.log("# Checkout the tag")
     session.run("git", "checkout", version, external=True, silent=True)
 
+    session.log("# Cleanup build/ before building the wheel")
+    if release.have_files_in_folder("build"):
+        shutil.rmtree("build")
```

Comment:
How about using Git to be consistent with other interactions with the system?
```suggestion
        session.run("git", "clean", "-fxd", "build/", external=True, silent=True)
```

Reply:
Even better, do it unconditionally:
```suggestion
    session.run("git", "clean", "-fxd", "build/", external=True, silent=True)
```

Reply:
+1

Is it safe/feasible to have a completely clean state before moving on? No just `build/`?

Reply:
I'm in for doing this in a follow up -- right now, I'll go ahead and merge this and go take a break for the day.

Reply:
@honnix dunno the specifics of Pip, @pradyunsg should answer this better. But in general, it should be safe to do this. I use this approach when building manylinux1 wheels which tends to leave some unwanted files on disk...

Reply:
Good job today. Appreciate it!

Reply:
*/me prepares that follow-up PR*